### ISSUE TITLE
opam upg recompiles conf-libX11.1 again and again

### ISSUE THREAD

**Issue by Fourchaux** [ocaml/opam#3995]
opam 2.0.5  - Linux x86_64 - OCaml 4.08.1+flambda

Installing/running **graphics** works well, but now, every _$ opam upg_ displays: 
```ocaml
$ opam upg
The following actions will be performed:
  ↻ recompile conf-libX11    1      [upstream changes]
  ↻ recompile graphics       4.08.1 [uses conf-libX11]
  ↻ recompile ocamlfind      1.8.1  [uses graphics]
...
  ↻ recompile utop           2.4.2  [uses ocamlfind]
===== ↻ 34 =====
Do you want to continue? [Y/n] 
```
Idem with OCaml 4.09.0+flambda
```ocaml
$ opam upg
The following actions will be performed:
  ↻ recompile conf-libX11    1      [upstream changes]
  ↻ recompile graphics       5.0.0  [uses conf-libX11]
  ↻ recompile ocamlfind      1.8.1  [uses graphics]
...
  ↻ recompile utop           2.4.2  [uses ocamlfind]
===== ↻ 32 =====
Do you want to continue? [Y/n] 
```
Any hint?

**Comment by dra27** [comment]
I've submitted a PR to workaround the issue, but there's also a bug in opam which I'm just working on

**Comment by Fourchaux** [comment]
:smile:

**Comment by rjbou** [comment]
Same problem occured in #4162 with `conf-dbm`

**Comment by dra27** [comment]
This should be permanently resolved with opam-file-format 2.1.x
